### PR TITLE
Ports cluwnes

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Ports the cluwne from Beestation (credits @qwertyquerty) (EDIT: originally from GoonStation).
It's mosty a punishment, the items cannot be removed. The code isn't fantastic so I'd appreciate review. 

This includes:

- admin smite
- dna mutation
- spawned items which turn you into a cluwne
- basic mob

<details>
<summary>images</summary>

This creature
Screenshot 2023-07-04 161453

A cluwned player
Screenshot 2023-07-04 184338

</details>

## Why It's Good For The Game
It really isn't. I'm pretty sure this brings a _curse_ upon the space station. Is that... Is that a honk in the distance!?
## Changelog
:cl:
add: Adds the cluwne mob. Honk honk! How terrifying.
admin: Players can now be cluwned...
/:cl:

### CODE REVIEW

**Comment by qwertyquerty** [comment]
Make sure you get floor cluwnes too

**Comment by LemonInTheDark** [comment]
it's gonna happen again I'm getting it in early

File: code/modules/mob/living/basic/cursed/cluwne.dm [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		speech_args[SPEECH_MESSAGE] = pick(
+			"HEEEENKKKKKK!!", \
+			"HONK HONK HONK HONK!!",\
+			"HONK HONK!!",\
+			"HOOOOOONKKKK!!", \
+			"HOOOOINKKKKKKK!!", \
+			"HOINK HOINK HOINK HOINK!!", \
+			"HOINK HOINK!!", \
+			"HOOOOOOIIINKKKK!!"\
+			)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/basic/cursed/cluwne.dm:80]: early returns

File: code/modules/mob/living/basic/cursed/cluwne.dm [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	playsound(src, 'sound/misc/honk_echo_distant.ogg', 50, 2)
+	var/newname = pick(GLOB.clown_names)
+	name = newname
+	real_name = newname
+	AddElement(/datum/element/waddling)
+	RegisterSignal(src, COMSIG_MOB_SAY, PROC_REF(handle_speech))
+
+/mob/living/basic/cluwne/Destroy()
+	. = ..()
+	UnregisterSignal(src, COMSIG_MOB_SAY)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/basic/cursed/cluwne.dm:44]: unneeded, signals clear on destroy

File: code/modules/mob/living/basic/cursed/cluwne.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/mob/living/basic/cursed/cluwne.dm:80]: also couldn't this just be a say() override?

File: code/modules/clothing/under/cluwne.dm [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	.=..()
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT(type))
+
+/obj/item/clothing/under/rank/civilian/cluwne/equipped(mob/living/carbon/user, slot)
+	if(!user.has_dna())
+		return
+	if(slot == ITEM_SLOT_ICLOTHING)
+		var/mob/living/carbon/player = user
+		player.dna.add_mutation(/datum/mutation/human/cluwne)
+	return ..()
```
Comment (LemonInTheDark) [inline code/modules/clothing/under/cluwne.dm:20]: isn't this `SHOULD_CALL_PARENT(TRUE)`, why is it safe to return if the owner has no dna? what is that meant to do?
This is a reaction proc, wouldn't that be an invalid state?

File: code/modules/clothing/shoes/cluwne.dm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	var/list/squeak_sound = list('sound/effects/footstep/clownstep1.ogg' = 1, 'sound/effects/footstep/clownstep2.ogg' = 1)
+
+/obj/item/clothing/shoes/cluwne/Initialize(mapload)
+	.=..()
+	LoadComponent(/datum/component/squeak, squeak_sound, 50, falloff_exponent = 20) //die off quick please
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT(type))
+
+/obj/item/clothing/shoes/cluwne/equipped(mob/user, slot)
+	if(!user.has_dna())
+		return
```
Comment (LemonInTheDark) [inline code/modules/clothing/shoes/cluwne.dm:21]: similar

File: code/modules/clothing/shoes/cluwne.dm [hunk 5]
```diff
@@ -0,0 +1,35 @@
+/obj/item/clothing/shoes/cluwne
+	desc = "The prankster's standard-issue clowning shoes. Damn, they're huge!"
+	name = "clown shoes"
+	icon_state = "clown"
+	inhand_icon_state = "clown_shoes"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	item_flags = DROPDEL
+	slowdown = SHOES_SLOWDOWN+1
```
Comment (LemonInTheDark) [inline code/modules/clothing/shoes/cluwne.dm:8]: why + 1 and not a new define?

File: code/modules/clothing/masks/cluwne.dm [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+/obj/item/clothing/mask/gas/cluwne/happy_cluwne/attack_self(mob/user)
+	voicechange = !voicechange
+	to_chat(user, span_notice("You turn the voice box [voicechange ? "on" : "off"]!"))
+	if(voicechange)
+		play_laugh()
+
+/obj/item/clothing/mask/gas/cluwne/happy_cluwne/equipped(mob/user, slot)
+	if(!ishuman(user))
+		return
```
Comment (LemonInTheDark) [inline code/modules/clothing/masks/cluwne.dm:79]: again

File: code/modules/clothing/masks/cluwne.dm [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+	var/mob/living/carbon/human/victim = user
+	if(slot == ITEM_SLOT_MASK && is_cursed)
+		log_admin("[key_name(victim)] was made into a cluwne by [src]")
+		message_admins("[key_name(victim)] got cluwned by [src]")
+		to_chat(victim, span_userdanger("The masks straps suddenly tighten to your face and your thoughts are erased by a horrible green light!"))
+		victim.dropItemToGround(src)
+		victim.cluwneify()
+		qdel(src)
```
Comment (LemonInTheDark) [inline code/modules/clothing/masks/cluwne.dm:87]: is it safe to call parent after a qdel here

File: code/modules/clothing/masks/cluwne.dm [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+		speech_args[SPEECH_MESSAGE] = pick(
+			"HEEEENKKKKKK!!", \
+			"HONK HONK HONK HONK!!",\
+			"HONK HONK!!",\
+			"HOOOOOONKKKK!!", \
+			"HOOOOINKKKKKKK!!", \
+			"HOINK HOINK HOINK HOINK!!", \
+			"HOINK HOINK!!", \
+			"HOOOOOOIIINKKKK!!"\
+			)
```
Comment (LemonInTheDark) [inline code/modules/clothing/masks/cluwne.dm:60]: early return

File: code/modules/clothing/masks/cluwne.dm [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+	var/delay = 15
+
+/obj/item/clothing/mask/gas/cluwne/Initialize(mapload)
+	.=..()
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT(type))
+
+/obj/item/clothing/mask/gas/cluwne/proc/play_laugh()
+	if(world.time - delay > last_sound)
+		playsound (src, pick('sound/items/SitcomLaugh1.ogg', 'sound/items/SitcomLaugh2.ogg', 'sound/items/SitcomLaugh3.ogg'), 30, 1)
+		last_sound = world.time
```
Comment (LemonInTheDark) [inline code/modules/clothing/masks/cluwne.dm:26]: you should use the cooldown macros `COOLDOWN_START`, `COOLDOWN_DECLARE`, etc

File: code/modules/clothing/masks/cluwne.dm [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+	item_flags = DROPDEL
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	modifies_speech = TRUE
+	w_class = WEIGHT_CLASS_SMALL
+	/// The voice change toggle.
+	var/voicechange = TRUE
+	/// The last time a sound was played.
+	var/last_sound = 0
+	/// The delay between sounds.
+	var/delay = 15
```
Comment (LemonInTheDark) [inline code/modules/clothing/masks/cluwne.dm:17]: Use SECONDS

File: code/modules/clothing/masks/cluwne.dm [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+	has_fov = FALSE
+	icon_state = "cluwne"
+	inhand_icon_state = "cluwne"
+	item_flags = DROPDEL
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	modifies_speech = TRUE
+	w_class = WEIGHT_CLASS_SMALL
+	/// The voice change toggle.
+	var/voicechange = TRUE
+	/// The last time a sound was played.
```
Comment (LemonInTheDark) [inline code/modules/clothing/masks/cluwne.dm:14]: what is time here?

File: code/modules/admin/verbs/cluwneplayer.dm [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+	Paralyze(3 SECONDS)
+	sleep(3 SECONDS)
+
+	to_chat(player_mind, span_honknosis("There is only the <b><i>Honkmother</i></b> now."))
+	sleep(1 SECONDS)
+
+	var/mob/living/basic/cluwne/newmob = new(get_turf(src))
+	player_mind.transfer_to(newmob)
+	if(key)  // afk (no mind)
+		newmob.key = key
```
Comment (LemonInTheDark) [inline code/modules/admin/verbs/cluwneplayer.dm:24]: why does this need to be nullchecked?
also what does mind here mean do you mean no key? one is a player's identity in the story of a round and the other is their direct connection and control over a mob

File: code/datums/mutations/cluwne.dm [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+	owner.ignite_mob()
+	owner.dna.add_mutation(/datum/mutation/human/cluwne)
+
+/mob/living/carbon/proc/cluwneify()
+	dna.add_mutation(/datum/mutation/human/cluwne)
+	emote("scream")
+	regenerate_icons()
+	visible_message(span_danger("[src]'s body glows green, the glow dissipating only to leave behind a cluwne formerly known as [src]!"), \
+					span_danger("Your brain feels like it's being torn apart, there is only the honkmother now."))
+	flash_act()
```
Comment (LemonInTheDark) [inline code/datums/mutations/cluwne.dm:57]: does this call `cluwne`? what is the difference between `cluwne` and `cluwneify`

**Comment by jlsnow301** [comment]
> Make sure you get floor cluwnes too

I'm not planning on adding this. This version is also a little more lighthearted, it's the sitcom laugh vs the maniacal cluwne laugh

File: code/datums/mutations/cluwne.dm [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+		var/mob/living/carbon/human/victim = owner
+		if(!istype(victim.w_uniform, /obj/item/clothing/under/rank/civilian/cluwne))
+			if(!victim.dropItemToGround(victim.w_uniform))
+				qdel(victim.w_uniform)
+			victim.equip_to_slot_or_del(new /obj/item/clothing/under/rank/civilian/cluwne(victim), ITEM_SLOT_ICLOTHING)
+		if(!istype(victim.shoes, /obj/item/clothing/shoes/cluwne))
+			if(!victim.dropItemToGround(victim.shoes))
+				qdel(victim.shoes)
+			victim.equip_to_slot_or_del(new /obj/item/clothing/shoes/cluwne(victim), ITEM_SLOT_FEET)
+		owner.equip_to_slot_or_del(new /obj/item/clothing/gloves/color/white(owner), ITEM_SLOT_GLOVES) // ditto
```
Comment (LemonInTheDark) [inline code/datums/mutations/cluwne.dm:30]: early returns
also how could the owner not be human isn't that checked in our parent proc's `on_acquiring`

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by nukularpower** [comment]
Man, I love you.   Its been a travesty to me that cluwnes arent in TGstation for as long as I've played here - I even spraypaint my clown character green to be cluwne themed.   HOWEVER... I do honestly dislike the validness.   Frankly, if I want to live as a cluwne, that shouldnt be your problem.   On Goonstation, the Chaplain can un-curse cluwnes by hitting them with their bible, allowing them to take off the cursed clothes - could that possibly be ported over with this?

I ask because then it would make sense that rather than just murdering a cluwne, you bring them to the Chaplain to be uncursed.   Thanks for making it, though, this is a great day

File: code/modules/mob/living/basic/cursed/cluwne.dm (hunk 2 above)
Comment (jlsnow301) [reply code/modules/mob/living/basic/cursed/cluwne.dm:44]: thanks,. i was wondering about this

File: code/modules/clothing/shoes/cluwne.dm (hunk 5 above)
Comment (jlsnow301) [reply code/modules/clothing/shoes/cluwne.dm:8]: doesn't seem to be how the define works, look at clown's shoes also, galoshes

Comment (LemonInTheDark) [reply code/modules/clothing/shoes/cluwne.dm:8]: `SHOES_SPEED_SLIGHT` (SHOES_SLOWDOWN - 1) exists, why not have the inverse?

File: code/datums/mutations/cluwne.dm (hunk 13 above)
Comment (jlsnow301) [reply code/datums/mutations/cluwne.dm:57]: i'll add more comments to these to make them more clear

**Comment by jlsnow301** [comment]
> HOWEVER... I do honestly dislike the validness. 

Same! It was sort of the running joke to kill them on sight, but I never liked it. They're harmless

> edit: will there be a wizard spell to cluwne curse people like on most stations?

I haven't added this, no plans for it (yet)

File: code/modules/clothing/shoes/cluwne.dm (hunk 5 above)
Comment (jlsnow301) [reply code/modules/clothing/shoes/cluwne.dm:8]: SHOES SPEED SLIGHT is not used anywhere btw

**Comment by Iamgoofball** [comment]
>Ports the cluwne from Beestation

>>>from Beestation

cluwnes are from GOON

**Comment by jlsnow301** [comment]
> cluwnes are from GOON

does this present a licensing issue?

**Comment by TheVekter** [comment]
> > cluwnes are from GOON
> 
> does this present a licensing issue?

Ah fuck here we go again

**Comment by san7890** [comment]
mark ready for review when ci is passing

**Comment by jlsnow301** [comment]
> mark ready for review when ci is passing

i'm not sure how to fix that, it's reordering tgstation.dme

**Comment by ChungusGamer666** [comment]
No floor cluwne? Sad
Guess that'll be up for me to PR

**Comment by itseasytosee** [comment]
are they going to be valid like goon cluwnes?

**Comment by jlsnow301** [comment]
> are they going to be valid like goon cluwnes?

I think that's more of a policy choice. I wouldn't like for them to be instantly valid tbh

**Comment by nukularpower** [comment]
Thoughts on making the cluwne curse removable via chaplain bible bash like on Goon?   To prevent the "valid" issue

**Comment by ATH1909** [comment]
I thought being valid on-sight was part of the point

**Comment by jlsnow301** [comment]
> Thoughts on making the cluwne curse removable via chaplain bible bash like on Goon? To prevent the "valid" issue.
> 
> meaning they lose the mutation and can take off their clothes, stop randomly going deaf and getting honk-blasted, etc

i'd like to see this implemented in a future iteration, i currently don't have the time to implement this

**Review by Wallemations — APPROVED** [review]
From a moral (_and potentially legal_) perspective I entirely detest this, but sadly from a spriting perspective I am forced to approve.

**Comment by Jacquerel** [comment]
Not to be the fun killer but I don't see much value in this.
I know it's like ss13 culture but it's not _tg_ ss13 culture. 
Turning people into a punishment mob which is automatically valid just provokes non-RP behaviour in people and frankly I just don't think the joke of a screaming and suffering clown which people beat to death in the corridors is really that funny either.

**Comment by Justice12354** [comment]
To be quite frank, I like the cluwnes and I would not mind them being valid at all, but, to me, their only appeal is the fact they are "soul" and I do not like the idea of porting "soulful" cultural content into our codebase. It just doesn't make sense. It's just like if I sprited something in a shitty way just so that it looked "soulful".

**Comment by Wallemations** [comment]
I agree with Jac
At most I could see these as a rare clown cytology thing or something, or a clown-ops version of romerol zombies

**Comment by jlsnow301** [comment]
no problem

**Comment by nukularpower** [comment]
Very sad to see this closed.   It was certainly possible to implement without them being valid for murder (as long as it was possible to cure, as I suggested.)    

Disappoint

**Pull request closed without merging**